File: Testing/Code for Testing/Basic/JS/M002_functions_test.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def add(a, b):
    return (a + b)

def multiply(a, b):
    return (a * b)

def factorial(n):
    if (n <= 1):
        return 1
    return (n * factorial((n - 1)))

def greet(name):
    print(('Hello, ' + name))

def main():
    print(add(3, 7))
    print(multiply(6, 8))
    print(factorial(5))
    print(factorial(10))
    greet('Tariq')
    print(add(multiply(3, 4), 5))
    return 0


if __name__ == "__main__":
    main()

File: Testing/Code for Testing/Basic/JS/M002_functions_test.js

function add(a, b) {
    return (a + b);
}

function multiply(a, b) {
    return (a * b);
}

function factorial(n) {
    if ((n <= 1)) {
        return 1;
    }
    return (n * factorial((n - 1)));
}

function greet(name) {
    console.log(("Hello, " + name));
}

function main() {
    console.log(add(3, 7));
    console.log(multiply(6, 8));
    console.log(factorial(5));
    console.log(factorial(10));
    greet("Tariq");
    console.log(add(multiply(3, 4), 5));
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
